src/cluster_legacy.c

```diff
@@ -5066,7 +5066,7 @@ void clusterCron(void) {
             if (!(node->flags & (CLUSTER_NODE_PFAIL | CLUSTER_NODE_FAIL))) {
                 node->flags |= CLUSTER_NODE_PFAIL;
                 update_state = 1;
-                if (server.cluster->size == 1 && clusterNodeIsVotingPrimary(myself)) {
+                if (clusterNodeIsVotingPrimary(myself)) {
```

Comment:
This change makes a lot of sense to me. I actually don't fully understand why we special-cased single-shard clusters in https://github.com/valkey-io/valkey/commit/b3aaa0a1362d229ba1ecd44629655f76c77304ec

Reply:
Maybe it was just to reduce the impact (to solve a specific problem (single shard) at the time). @bentotten would you be able to take a review?

Reply:
For single-shard clusters, there is no need to wait for a quorum of 1, so we can proceed to directly fail the node here and not wait for the original gossip-logic [(for reference)](https://github.com/valkey-io/valkey/blame/unstable/src/cluster_legacy.c#L2256). Unfortunately prior to that, a single-shard cluster would never properly mark a replica as failed.

I believe we still need to wait to achieve a quorum from gossip to mark this node as truly failed, no? If the node uses the information it has from "old gossip" during each cluster cron run, wont this increase the number of calls to this function without increasing the accuracy of the failure report?

Reply:
Though the benefit of the node being able to mark the replica as failed as soon as it knows it cant reach said replica in the last-needed-vote scenario also makes sense - @madolson I know you did the initial investigation for this change, if you have any insights

Reply:
right, thanks for the input, that is new to me. clusterNodeFailureReportsCount will cleanup the old failure report, so i think (hope) it can avoid the old gossip information risk. But you do make a good point, let see if others have a new thought.

Reply:
Sorry, I don't have any useful input about this scenario. You know it better than me.

Reply:
no worry, it is always good to have an extra eyes on it.